modules/data_warehouse/bigquery.tf

```diff
@@ -0,0 +1,334 @@
+  ]
+}
+
+# # Add Sample Queries
+resource "google_bigquery_routine" "sp_sample_queries" {
+  project         = module.project-services.project_id
+  dataset_id      = google_bigquery_dataset.ds_edw.dataset_id
+  routine_id      = "sp_sample_queries"
+  routine_type    = "PROCEDURE"
+  language        = "SQL"
+  definition_body = templatefile("${path.module}/src/sql/sp_sample_queries.sql", { project_id = module.project-services.project_id })
+
+  depends_on = [
+    google_bigquery_table.tbl_edw_taxi,
+  ]
+}
+
+# # Add Bigquery ML Model
+resource "google_bigquery_routine" "sp_bigqueryml_model" {
+  project         = module.project-services.project_id
+  dataset_id      = google_bigquery_dataset.ds_edw.dataset_id
+  routine_id      = "sp_bigqueryml_model"
+  routine_type    = "PROCEDURE"
+  language        = "SQL"
+  definition_body = templatefile("${path.module}/src/sql/sp_bigqueryml_model.sql", { project_id = module.project-services.project_id })
+
+  depends_on = [
+    google_bigquery_table.tbl_edw_taxi,
+  ]
+}
+
+# # Add Translation Scripts
+resource "google_bigquery_routine" "sp_sample_translation_queries" {
+  project         = module.project-services.project_id
+  dataset_id      = google_bigquery_dataset.ds_edw.dataset_id
+  routine_id      = "sp_sample_translation_queries"
+  routine_type    = "PROCEDURE"
+  language        = "SQL"
+  definition_body = templatefile("${path.module}/src/sql/sp_sample_translation_queries.sql", { project_id = module.project-services.project_id })
+
+  depends_on = [
+    google_bigquery_table.tbl_edw_taxi,
+  ]
+}
+
+# Add Scheduled Query
+# # Set up DTS permissions
+resource "google_project_service_identity" "bigquery_data_transfer_sa" {
+  provider = google-beta
+  project  = module.project-services.project_id
+  service  = "bigquerydatatransfer.googleapis.com"
+}
+
+# # Grant the DTS service account access
+resource "google_project_iam_member" "dts_service_account_roles" {
+  for_each = toset([
+    "roles/bigquerydatatransfer.serviceAgent",
+  ])
+
+  project = module.project-services.project_id
+  role    = each.key
+  member  = "serviceAccount:${google_project_service_identity.bigquery_data_transfer_sa.email}"
+}
+
+# Create specific service account for DTS Run
+# # Set up the DTA service account
+resource "google_service_account" "dts" {
+  project      = module.project-services.project_id
+  account_id   = "cloud-dts-sa-${random_id.id.hex}"
+  display_name = "Service Account for Data Transfer Service"
+}
+
+# # Grant the DTS Specific service account access
+resource "google_project_iam_member" "dts_roles" {
+  for_each = toset([
+    "roles/bigquery.user",
+    "roles/bigquery.dataEditor",
+  ])
+
+  project = module.project-services.project_id
+  role    = each.key
+  member  = "serviceAccount:${google_service_account.dts.email}"
+}
+
+# # Grant the DTS specific service account Token Creator to the DTS Service Identity
+resource "google_service_account_iam_binding" "dts_token_creator" {
```

Comment:
nit: prefer non authoritative binding
https://cloud.google.com/docs/terraform/best-practices-for-terraform#iam

Reply:
added bug to track